### ISSUE TITLE
Incoming call notifications not received in IOS

### ISSUE THREAD

**Issue by aparnakasi16** [twilio/twilio-voice-react-native#457]
# Issue

## Pre-submission Checklist
- [ ] I have verified that the issue occurs with the latest release and is not marked as a known issue in the [CHANGELOG.md](https://github.com/twilio/twilio-voice-react-native/blob/main/CHANGELOG.md).
- [x] I reviewed the [Common Issues](https://github.com/twilio/twilio-voice-react-native/blob/main/COMMON_ISSUES.md) and open GitHub issues and verified that this report represents a potentially new issue.
- [x] I am not sharing any [Personally Identifiable Information (PII)](https://www.twilio.com/docs/glossary/what-is-personally-identifiable-information-pii)
       or sensitive account information (API keys, credentials, etc.) when reporting this issue.

## Description 
I have successfully registered to twilio push notification with VOIP registry token. But my application is unable to receive any notification on incoming call. Even in logs, I am unable to notice callInvite being triggered. No action is visible on Xcode Logs.
CallSid : [CA8627ad637c097de70f5a27c513c4b871]

## Reproduction Steps
1. run app on iPad
2. login and register successful
3. Outgoing works fine
4. No action on incoming

## Expected Behavior
On making an incoming call to my twilio number, incoming call notification was expected. 

## Actual Behavior
Application stays idle. No action or logs. No notification is received

## Reproduction Frequency
Multiple attempts

## Screenshots
Xcode logs : 
TRACE:Twilio:[Platform](0x16dbab000) : Registering for Twilio push notifications
Access Token: 'access token'
VoIP Registry Token: 'VOIP registry token'
DEBUG:Twilio:[Platform](0x16dbab000) : Using default home region
INFO:Twilio:[Platform](0x16dbab000) : Background task started for registration/unregistration, identifer = 12
DEBUG:Twilio:[Platform](0x16dbab000) : Register for push notifications.
Access token saved to AsyncStorage
DEBUG:Twilio:[Platform](0x16db1f000) : Completed registration request.
DEBUG:Twilio:[Platform](0x16db1f000) : Registration location: https://ers.us1.twilio.com/v1/registrations/xxxxxx
INFO:Twilio:[Platform](0x1fe05a480) : Inside register:deviceToken:completion:, registered for Twilio push notifications.
'voice registration result', { status: 'fulfilled', value: undefined }
registration/register/fulfilled
loginAndRegister/fulfilled
DEBUG:Twilio:[Platform](0x1fe05a480) : Ending registration/unregistration background task, identifier = 12
DEBUG:Twilio:[Platform](0x1fe05a480) : Registration/unregistration background task ended

## Software and Device Information
**Please complete the following information.**
 - Device: iPad 9th Gen
 - OS: IOS 18.1.1
 - React version: 18.2.0
 - React Native version: 0.72.17
 - Node version: v22.6.0
 - npm or yarn version:10.8.2

## Additional Context
When an incoming call comes in, do I need to add a webhook with any info? if I am missing.
Is there any backend code required to trigger notification

**Comment by mhuynh5757** [comment]
Hi @aparnakasi16 thanks for reaching out. Is this a real physical iPad or a simulated device?

**Comment by mhuynh5757** [comment]
Also, have you invoked `voice.initializePushRegistry` before attempting to register for incoming calls?

**Comment by aparnakasi16** [comment]
Hi @mhuynh5757 , Yes it is a physical iPad and I have also invoked voice.initializePushRegistry at the very beginning of my app. I'am able to manually send voip notifications through icloud, so there seems to be no issue with my voip certificate and integration as well. Yet I'm unable to receive any notification on incoming call. Is there something I'm missing in my backend ? Probably in When an incoming call comes in weebhook ?

**Comment by aparnakasi16** [comment]
Hi @mhuynh5757 , Thank you for replying . I am able to receive incoming call notification. Changed my backend endpoint when an incoming call comes to make a client call. But when the app is in killed state, I am unable to receive any notification. Could you please give me an input on this ?

**Comment by mhuynh5757** [comment]
@aparnakasi16 you were able to perform `voice.register(token)` successfully and were receiving incoming calls when the app is foregrounded? And then, let's say you swipe it away from the multitasking view to "kill" the app? And at that point you no longer receive incoming calls?

**Comment by tamdvyounetgroup** [comment]
=> follow here : https://github.com/tamdvyounetgroup/twilio-voice-react-native

**Comment by aparnakasi16** [comment]
> @aparnakasi16 you were able to perform `voice.register(token)` successfully and were receiving incoming calls when the app is foregrounded? And then, let's say you swipe it away from the multitasking view to "kill" the app? And at that point you no longer receive incoming calls?

Yes, that's the case. Receiving incoming call when app is in foreground as well as backgrounf. But when the app is killed, I'm not able to receive.

**Comment by mhuynh5757** [comment]
I see. And since the app is killed, it will be hard to get Xcode debug logs from your application. If you can try, will Xcode show any crashes or logs for this scenario? I understand if not.

**Comment by tamdvyounetgroup** [comment]
> > @aparnakasi16 you were able to perform `voice.register(token)` successfully and were receiving incoming calls when the app is foregrounded? And then, let's say you swipe it away from the multitasking view to "kill" the app? And at that point you no longer receive incoming calls?
> 
> Yes, that's the case. Receiving incoming call when app is in foreground as well as background. But when the app is killed, I'm not able to receive.

Please follow my fork.

**Comment by mhuynh5757** [comment]
Hey all, I'd like to close this issue as stale for now. Please feel free to reach out if you need further assistance.

**Comment by usmanabid94** [comment]
I am facing an issue with getting incoming calls on IOS here the code block for it what could be wrong here 
 // Handle incoming calls
    useEffect(() => {
        if (token) {
            const instance = new Voice();
            console.log("aslam")
            instance.initializePushRegistry()
            // Allow incoming calls
            instance.register(token);
            instance.on('incoming', (call) => {
                // Display incoming call UI
                // Provide options to accept or reject the call
                console.log('Incoming call invite received2221211:', call);
            });

            // Handle incoming calls
            instance.on(Voice.Event.CallInvite, (callInvite: any) => {
                console.log('Incoming call invite received:', callInvite);
                if (callInvite) {
                    callInvite.accept(); // Accept the incoming call
                    console.log('Call accepted');
                } else {
                    console.error('No callInvite received');
                }
            });

            instance.on(Voice.Event.Registered, () => {
                console.log('Device registered successfully for incoming calls.');
            });
            instance.on(Voice.Event.Error, error => {
                console.error('Error during call event:', error);
            });

            setVoiceInstance(instance);

        }
    }, [token]);
